socs/agents/galildmc_axis_controller/drivers.py

```diff
@@ -0,0 +1,504 @@
+        the appropriate commutation angle. Cannot command with BZ unless BA
+        and BM commands are sent first.
+
+        """
+        resp = self.galil_command(command=f'BZ{axis}={val};')
+        return resp
+
+    def define_position(self, axis, val=0):
+        """
+        Redefines current axis position to user specified value.
+        Useful for homing procedure. Default is 0.
+
+        """
+        resp = self.galil_command(command=f'DP{axis}={val};')
+        return resp
+
+    def disable_limit_switch(self, axis):
+        """
+        Disable limit switch detection on a given axis (LDx=3).
+
+        """
+        resp = self.galil_command(command=f'LD{axis}=3;')
+        return resp
+
+    def set_limitswitch_polarity(self, pol=1):
+        """
+        CN -1 means active low, CN +1 is active high. Default is  1
+        (active high).
+
+        """
+        resp = self.galil_command(command=f'CN {pol};')
+        return resp
+
+    def stop_motion(self, axis):
+        """
+        Stop motion.
+
+        """
+        cmd = f"ST {axis};"
+        resp = self.galil_command(command=cmd)
+        return resp
+
+    def set_gearing(self, order):
+        """
+        Set gearing: order is order of opertions in string: ',A,,C'.
+
+        """
+        resp = self.galil_command(command=f"GA {order};")
+        return resp
+
+    def set_gearing_ratio(self, order):
+        """
+        Set gearing ratios, e.g. GR -1,1 for axes B and D.
+
+        """
+        resp = self.galil_command(command=f"GR {order};")
+        return resp
+
+    def jog_axis(self, axis, speed):
+        """
+        Set jog speed for axis. Does not begin motion, just sets up speed
+        for when ready to begin motion.
+
+        """
+        cmd = f"JG{axis}={speed};"
+        resp = self.galil_command(command=cmd)
+        return resp
+
+    def set_speed(self, axis, speed):
+        """
+        Set speed for axis for use in commands like PR/PA (relative and absolute
+        position). Does not begin motion, just defines speed for when
+        ready to begin motion. Units of speed is in raw encoder units (counts/s).
+
+        """
+        cmd = f"SP{axis}={speed};"
+        resp = self.galil_command(command=cmd)
+        return resp
+
+    def set_motor_state(self, axis, state):
+        """
+        Enable or disable a motor, then verify its state (0='on', 1='off').
+
+        """
+        state = state.lower().strip()
+        if state not in ('enable', 'disable'):
+            raise ValueError("state must be 'enable' or 'disable'.")
+
+        # --- Execute command ---
+        if state == 'enable':
+            self.galil_command("SH", axis=axis)
+        elif state == 'disable':
+            self.galil_command("MO", axis=axis)
+
+        # --- Allow controller to settle ---
+        time.sleep(1.0)
+
+        # --- Query new state ---
+        status, human_state = self.get_motor_state(axis=axis)
```

Comment:
In most/all of the other "set" functions, you don't call its related "get" function within it. I comment on this in my overall review comment, but it's best to standardize to one or the other.